persistent/Database/Persist/Quasi.hs

```diff
@@ -181,9 +432,22 @@ empty _          = False
 
 -- | A line.  We don't care about spaces in the middle of the
 -- line.  Also, we don't care about the ammount of indentation.
-data Line = Line { lineIndent :: Int
-                 , tokens     :: [Text]
-                 }
+data Line' f
+    = Line
+    { lineIndent   :: Int
+    , tokens       :: f Text
+    }
```

Comment:
generalizing `Line'` allowed me to write safe functions, by guaranteeing that a `tokens` was `NonEmpty`